src/main/java/org/openlmis/requisition/service/RequisitionService.java

```diff
@@ -398,20 +412,77 @@ public Requisition reject(UUID requisitionId) {
   }
 
   /**
-   * Checks if given user has permission to approve requisition, if not exception is thrown.
+   * Performs several validation checks to ensure that the given requisition can be approved.
+   * It makes sure that the user has got rights to approve the requisition, that the requisition
+   * exists and that it has got correct status to be eligible for approval.
    *
-   * @param programId         UUID of program that requisition is assigned to
-   * @param supervisoryNodeId UUID of supervisory node where requisition have to be approved
-   * @param userId            UUID of user that wants to approve requisition
+   * @param requisition the requisition to verify
+   * @param requisitionId requisition ID for which the request was made
+   * @param userId the UUID of the user approving the requisition
+   * @return ValidationResult instance containing the outcome of this validation
    */
-  public void checkIfCanApproveRequisition(UUID programId, UUID supervisoryNodeId, UUID userId) {
-    RightDto right = rightReferenceDataService.findRight(RightName.REQUISITION_APPROVE);
+  public ValidationResult validateCanApproveRequisition(Requisition requisition, UUID requisitionId,
+                                                        UUID userId) {
+
+    ValidationResult permissionCheck = permissionService.canApproveRequisition(requisitionId);
+    if (permissionCheck.hasErrors()) {
+      return permissionCheck;
+    }
 
+    if (requisition == null) {
+      return ValidationResult.notFound(
+          MessageKeys.ERROR_REQUISITION_NOT_FOUND, requisitionId);
+    }
+
+    if (configurationSettingService.getSkipAuthorization() ? requisition.getStatus()
+        != RequisitionStatus.SUBMITTED : !requisition.isApprovable()) {
```

Comment:
Can this logic be included in requisition.isApprovable method that will have param skipAuthorization?